Core/Core/CourseSync/CourseSyncProgress/ViewModel/CourseSyncProgressViewModel.swift

```diff
@@ -0,0 +1,112 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import CombineExt
+import SwiftUI
+
+class CourseSyncProgressViewModel: ObservableObject {
+    enum State {
+        case loading
+        case data
+        case error
+    }
+
+    // MARK: - Output
+
+    @Published public private(set) var state = State.loading
+    @Published public private(set) var cells: [Cell] = []
+
+    public let labels = (
+        noCourses: (
+            title: NSLocalizedString("No Courses", comment: ""),
+            message: NSLocalizedString("Your courses will be listed here, and then you can make them available for offline usage.", comment: "")
+        ),
```

Comment:
I think we won't be able to reach this state. We can only check the sync progress if there's something we are syncing so there will be at least one course.